src/uct/ib/base/ib_iface.c

```diff
@@ -272,6 +292,20 @@ static ucs_status_t uct_ib_iface_init_pkey(uct_ib_iface_t *iface,
     return UCS_ERR_INVALID_PARAM;
 }
 
+#if HAVE_DECL_IBV_LINK_LAYER_ETHERNET
+static int uct_ib_iface_is_gid_raw_empty(uint8_t *gid_raw)
+{
+    int i;
+
+    for (i = 0; i < 16; i++) {
```

Comment:
simpler and more efficient: 
return (*(uint64_t*)gid_raw == 0) && (*(uint64_t*)(gid_raw + 8) == 0)